plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/providedservices/pde_mgmt/routehandlers/GetAllPlantDescriptions.java

```diff
@@ -0,0 +1,161 @@
+            .name(SORT_FIELD)
+            .legalValues(ID, CREATED_AT, UPDATED_AT)
+            .build();
+
+        final StringParameter directionParam = new StringParameter.Builder()
+            .name(DIRECTION)
+            .legalValues(ASC, DESC)
+            .defaultValue(ASC)
+            .build();
+        final BooleanParameter activeParam = new BooleanParameter.Builder()
+            .name(ACTIVE)
+            .build();
+
+        final List<QueryParameter> acceptedParameters = List.of(pageParam, sortFieldParam, directionParam, activeParam);
+
+        final QueryParamParser parser;
+
+        try {
+            parser = new QueryParamParser(null, acceptedParameters, request);
+        } catch (final ParseError error) {
+            response
+                .status(HttpStatus.BAD_REQUEST)
+                .body(ErrorMessage.of(error.getMessage()), CodecType.JSON);
+            logger.error("Encountered the following error(s) while parsing an HTTP request: " + error.getMessage());
+            return Future.success(response);
+        }
+
+        List<PlantDescriptionEntryDto> entries = pdTracker.getEntries();
+
+        final Optional<String> sortField = parser.getValue(sortFieldParam);
+        if (sortField.isPresent()) {
+            final String sortDirection = parser.getRequiredValue(directionParam);
+            final boolean sortAscending = (ASC.equals(sortDirection));
+            switch (sortField.get()) {
+                case ID:
+                    PlantDescriptionEntry.sortById(entries, sortAscending);
+                    break;
+                case CREATED_AT:
+                    PlantDescriptionEntry.sortByCreatedAt(entries, sortAscending);
+                    break;
+                case UPDATED_AT:
+                    PlantDescriptionEntry.sortByUpdatedAt(entries, sortAscending);
+                    break;
+                default:
+                    // We should never reach this case, since the sortField
+                    // param has been validated by the parser.
+                    throw new AssertionError("Encountered the invalid sort field '" + sortField + "'.");
+            }
+        }
+
+        final Optional<Integer> page = parser.getValue(pageParam);
+        if (page.isPresent()) {
+            final int itemsPerPage = parser.getRequiredValue(itemPerPageParam);
+            final int from = Math.min(page.get() * itemsPerPage, entries.size());
+            final int to = Math.min(from + itemsPerPage, entries.size());
+
+            entries = entries.subList(from, to);
+        }
+
+        final Optional<Boolean> active = parser.getValue(activeParam);
+        if (active.isPresent()) {
+            PlantDescriptionEntry.filterByActive(entries, active.get());
+        }
+
+        PlantDescriptionEntryListDto result = new PlantDescriptionEntryListDto.Builder()
+            .data(entries)
+            .count(entries.size())
```

Comment:
This count calculation works differently than in the similar endpoints of the other core systems. For example, in Service Registry's /mgmt/systems endpoint the count doesn't contain the size of the data, but the total number of records AFTER filtering but BEFORE paging.

You should move active filtering before the paging, then store the size of the entries in a local variable. And use this variable when you build the result.

Reply:
Fixed